app/src/ui/history/drag-and-drop-intro.ts

```diff
@@ -0,0 +1,39 @@
+/** Type of the drag and drop intro popover. Each value represents a feature. */
+export enum DragAndDropIntroType {
+  CherryPick = 'cherry-pick',
+  Squash = 'squash',
+  Reorder = 'reorder',
+}
+
+/** Structure to describe the drag and drop intro popover. */
+export type DragAndDropIntro = {
+  readonly title: string
+  readonly body: string
+}
+
+/** List of all available drag and drop intro types. */
+export const AvailableDragAndDropIntroKeys = Object.values(
+  DragAndDropIntroType
+) as ReadonlyArray<DragAndDropIntroType>
+
+/** Map with all available drag and drop intros. */
+export const AvailableDragAndDropIntros: Record<
+  DragAndDropIntroType,
+  DragAndDropIntro
+> = {
+  [DragAndDropIntroType.CherryPick]: {
+    title: 'Cherry-pick!',
+    body:
+      'Copy commits to another branch by dragging and dropping them onto a branch in the branch menu, or by right clicking on a commit.',
+  },
+  [DragAndDropIntroType.Squash]: {
+    title: 'Squash!',
+    body:
+      'Squash commits by dragging and dropping them onto another commit, or by right clicking on multiple commits.',
+  },
+  [DragAndDropIntroType.Reorder]: {
+    title: 'Reorder!',
+    body:
+      'Reorder commits to tidy up your history by dragging and dropping them to a different position.',
+  },
+}
```

Comment:
Not sure how much of this is considered bad practice… I just wanted to have:
1. A list of all the available intro types (so that I could easily grab how many there are, and see what haven't been seen yet)
2. A dictionary `type - title+body` where I could quickly get the title and body given an intro type.

I'm more than open to suggestions to improve or clean this up!

Reply:
Is a little different than what we seem to do in this app... but this is exactly how one team I was on did validation messages. It is fine to me. 🤷